src/lib/ContractComponents/baseERC721.ts

```diff
@@ -81,8 +82,15 @@ export default class BaseERC721 {
     }
 
     try {
-      let options = { gasLimit: GAS_LIMIT };
-      options = addGasPriceToOptions(options, params.gas);
+      const chainId = await this.contractDeployed.signer.getChainId();
+      let options;
+      // If Polygon mainnet, set up options propperly to avoid underpriced transaction error
+      if (chainId === 137)
```

Comment:
It might be convenient to define a constant class with common chain IDs:

`./src/lib/constants.ts`:
```typescript
export class ChainIDs {
  static readonly ETH_MAINNET = 1;
  static readonly POLYGON_MAINNET = 137;
  ...
}
```

Reply:
hey @bergeron I just used the const Chains constant that it's in availableChains.ts file